src/Storages/ObjectStorage/S3/Configuration.cpp

```diff
@@ -74,7 +127,9 @@ static const std::unordered_set<std::string_view> optional_configuration_keys =
     "max_single_part_upload_size",
     "max_connections",
     "expiration_window_seconds",
-    "no_sign_request"
+    "no_sign_request",
+    "partitioning_style",
```

Comment:
I was doubting if "partition_strategy" might be a better name, and then I saw that you have a class `PartitionStrategy` that basically represents what is controlled by this setting, and now I am certain.